app/Utils/HtmlEngine.php

```diff
@@ -277,14 +281,17 @@ public function buildEntityDataArray(): array
             $data['$credit.custom4'] = &$data['$quote.custom4'];
 
             if ($this->entity->project) {
-                $data['$project.name'] = ['value' => $this->entity->project->name, 'label' => ctrans('texts.project')];                
-                $data['$invoice.project'] = &$data['$project.name'];
-                $data['$quote.project'] = &$data['$project.name'];
-
+                $data['$project.name'] = ['value' => $this->entity->project->name, 'label' => ctrans('texts.project')];
+            } else {
+                $data['$project.name'] = ['value' => '', 'label' => ''];
             }
+            $data['$invoice.project'] = &$data['$project.name'];
+            $data['$quote.project'] = &$data['$project.name'];
 
             if ($this->entity->vendor) {
                 $data['$invoice.vendor'] = ['value' => $this->entity->vendor->present()->name(), 'label' => ctrans('texts.vendor_name')];
+            } else {
+                $data['$invoice.vendor'] = ['value' => '', 'label' => ''];
```

Comment:
instead of the if / else here, just define these empty entries at the top of the file somewhere, that way they will only be overwritten if the name exists